src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -240,7 +257,14 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
 
             // Read the first vector unaligned.
 
-            currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
+            if (Sse2.IsSupported)
+            {
+                currentMask = (ulong)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
+            }
+            else
```

Comment:
Likewise, replace with `else if (AdvSimd.Arm64.IsSupported) { ... } else { throw new PlatformNotSupportedException() }` to catch potential future bugs.